lib/symboldatabase.cpp

```diff
@@ -2712,6 +2712,10 @@ void SymbolDatabase::addClassFunction(Scope **scope, const Token **tok, const To
                             const Token *closeParen = (*tok)->next()->link();
                             if (closeParen) {
                                 if (Token::Match(closeParen, ") noexcept| = default ;") ||
+                                    Token::Match(closeParen, ") . %type% = default ;") ||
+                                    Token::Match(closeParen, ") . %type% & = default ;") ||
+                                    Token::Match(closeParen, ") . const %type% & = default ;") ||
+                                    Token::Match(closeParen, ") . const %type% = default ;") ||
```

Comment:
@danmar I'm pretty sure this can be done in a better way right?
and it also does not cover all cases yet.
So I'm hoping I will get some help here, please.

Reply:
well a start could be:

    Token::Match(closeParen, ") . const| %type% & = default ;"))

but not sure if that is the ultimate solution, maybe we have some code to handle trailing return type and stuff..

Reply:
imho, the `Token::Match` pattern I provided could be a quick fix I could merge that..

But there is some copy/paste problems in the SymbolDatabase. Maybe we should have some utility function to parse function attributes . There is a loop in `Function::Function` below a comment `// parse function attributes` that looks promising. I am spontanously wondering if that code can be broken out and reused somehow.

Reply:
another way is to reuse Tokenizer::isFunctionHead.. something like:

    const Token *endtok = mTokenizer->isFunctionHead(closeParent->link(), ";");
    if (endtok && Token::simpleMatch(endtok->tokAt(-2), "= default ;"))
    .....

.. well we have some options. I believe a small fix to start with, with a proper test and a `Token::Match` is a good first step.

Reply:
Alright, thank you I will clean it up and add a proper test for this.
maybe I'll find the time tomorrow evening